676 allow no regexp for parent in FASTA importer

bradfordcondon opened the pull request:
<!--- Thank you for contributing! -->
<!--- Provide a general summary of your changes in the Title above -->
<!--- See our Contribution Guidelines here:
          https://github.com/tripal/tripal/blob/7.x-3.x/CONTRIBUTING.md -->


<!---  Please set the header below based on the PR type:
# Documentation  --->

# New Feature

Issue #676 

## Description
The FASTA importer adds a relationship to another feature if you specify the type and a regexp.  however a regexp is not necessary if the parent feature name is identical to the feature you are importing.

This simple PR allows a form to be submitted with a parent feature type and relationship type but no regexp.  And, if this is so, it will search for the parent with the same uniquename.  This feature is found in other importers as discussed in #676.


## Testing?
As a bonus i wrote a test case for this feature.  We insert two mRNAs.  then we load polypeptides with the same names and verify that the importer links them (that the mRNA has a feature_relationship record)

bradfordcondon committed:
allow a relationship type and parent type to be specified with no regexp.  if so, then the names should match

bradfordcondon committed:
add tests for fasta importer

bradfordcondon committed:
also add to form instructions

bradfordcondon commented:
ill look into the failed test today, it passes locally...


oops, i added a join and forgot you cant chain joins.  should be fixed.

bradfordcondon committed:
oops dont chain joins

bradfordcondon commented on tripal_chado/includes/TripalImporter/FASTAImporter.inc:
```diff
@@ -536,7 +535,7 @@ class FASTAImporter extends TripalImporter {
 
         // Get the relationship subject
-         $subject = "";
+         $subject = $name ? $name : $uname;
```
should this actually be $uname or null? or instead of $name or $uname?

laceysanderson replied:
I think it should be 
```suggestion
         $subject = $uname ? $uname : "";
```
based on https://github.com/tripal/tripal/pull/738/files#diff-8006ac719a97e82d92833b60886802a2L831 where the query looking up the parent/subject for the relationship checks the uniquename column.

laceysanderson reviewed (commented):
Test passes locally for me 👍

laceysanderson committed:
Update tripal_chado/includes/TripalImporter/FASTAImporter.inc

I think so too!  Forgot to come back to this after I raised the question.  Thank you!

Co-Authored-By: bradfordcondon <bradford.condon@gmail.com>

bradfordcondon commented:
the second review for this PR does not need to be functional.

spficklin commented:
Non-functional code review looks good. I merged it.
